tables/v1/automations.sql

```diff
@@ -0,0 +1,31 @@
+SET SEARCH_PATH TO v1;
+
+CREATE TABLE IF NOT EXISTS automations
+(
+    id               SERIAL                   NOT NULL UNIQUE,
+    name             TEXT                     NOT NULL,
+    slug             TEXT                     NOT NULL,
+    integration_name TEXT                     NOT NULL,
+    callable         TEXT                     NOT NULL,
+    categories       automation_category_type[],
+    created_at       TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    created_by       TEXT                     NOT NULL DEFAULT 'system',
+    last_modified_at TIMESTAMP WITH TIME ZONE,
+    last_modified_by TEXT,
+    PRIMARY KEY (slug, integration_name),
```

Comment:
I'm curious why `id` isn't the PK? Is there a benefit to using these columns instead?

Reply:
I've been trying to use natural keys whenever possible and introducing surrogate keys (the `id` field) only when I need them. This makes the URLs cleaner and _a lot_ easier to use when you are `curl`'ing around. The only reason that I added the `id` field was that replicating the natural key pair through association tables that only have two columns is painful.